Neos.Flow/Classes/Mvc/Routing/RouteConfiguration.php

```diff
@@ -0,0 +1,120 @@
+<?php
+namespace Neos\Flow\Mvc\Routing;
+
+/*
+ * This file is part of the Neos.Flow package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Configuration\ConfigurationManager;
+use Neos\Flow\Mvc\Exception\InvalidRouteSetupException;
+
+/**
+ * The default router configuration
+ *
+ * @Flow\Scope("singleton")
+ */
+final class RouteConfiguration
+{
+    /**
+     * Array of routes to match against
+     *
+     * @var Routes
+     */
+    private Routes $routes;
+
+    /**
+     * true if route object have been created, otherwise false
+     *
+     * @var boolean
+     */
+    private bool $routesCreated = false;
+
+    public function __construct(
+        private readonly ConfigurationManager $configurationManager
+    ) {
+    }
+
+    /**
+     * Returns a list of configured routes
+     *
+     * @return Routes
+     */
+    public function getRoutes(): Routes
+    {
+        if ($this->routesCreated === false) {
+            $this->createRoutesFromConfiguration();
+        }
+        return $this->routes;
+    }
+
+    /**
+     * Creates \Neos\Flow\Mvc\Routing\Route objects from the injected routes
+     * configuration.
+     *
+     * @return void
+     * @throws InvalidRouteSetupException
+     */
+    protected function createRoutesFromConfiguration(): void
```

Comment:
```suggestion
    private function createRoutesFromConfiguration(): Routes
```

Reply:
It does not return anything

> I still like the direction, but I stumbled upon a major issue: This change removes the lazy loading & parsing of routes

Please but some more words to it, I really want to understand it, but don't at the moment 🙃